File: Python/chapitre4/Exercice1.py
import os
r, w = os.pipe()
pid = os.fork()
if pid == 0:
    os.close(r)
    mot = input("Mot: ")
    os.write(w, mot.encode() + b"\0")
    os._exit(0)
os.close(w)
mot = os.read(r, 256).decode().strip("\0")
with open("fichier.txt") as f:
    contenu = f.read().split()
trouve = 1 if mot in contenu else 0
print(trouve)
os.waitpid(pid, 0)

File: C/chapitre4/Exercice1.c
#include <stdio.h>
#include <string.h>
#include <unistd.h>
#include <sys/wait.h>

int main(void) {
    int tube[2];
    pipe(tube);
    pid_t pid = fork();
    if (pid == 0) {
        close(tube[0]);
        char mot[256];
        printf("Mot: ");
        scanf("%s", mot);
        write(tube[1], mot, strlen(mot) + 1);
        close(tube[1]);
        _exit(0);
    }
    close(tube[1]);
    char mot[256];
    read(tube[0], mot, sizeof(mot));
    FILE *f = fopen("fichier.txt", "r");
    char buf[256];
    int trouve = 0;
    while (fscanf(f, "%s", buf) != EOF) {
        if (strcmp(buf, mot) == 0) {
            trouve = 1;
        }
    }
    fclose(f);
    printf("%d\n", trouve);
    wait(NULL);
    return 0;
}